activerecord/test/cases/excluding_test.rb

```diff
@@ -22,6 +22,16 @@ def test_result_set_does_not_include_collection_of_excluded_records
     assert_not_includes relation, posts(:thinking)
   end
 
+  def test_result_set_does_not_include_collection_of_excluded_records_from_a_query
+    query = Post.where(id: @post)
+
+    assert_sql(/SELECT #{ActiveRecord::Base.connection.quote_table_name("posts.id")} FROM #{ActiveRecord::Base.connection.quote_table_name("posts")}/) do
```

Comment:
Nitpick: technically, it would be good practice to use `Regexp.escape` here.  Since this line is already rather verbose, we could probably omit the part after `FROM`.

```suggestion
    assert_sql(/SELECT #{Regexp.escape Post.connection.quote_table_name("posts.id")} FROM/) do
```